### ISSUE TITLE
Expected one of [ \\t\\r\\n], \”#\”, \”input\”, \”filter\”, \”output\” at line 1, column 1 (byte 1)

### ISSUE THREAD

**Issue by ZoolYe** [elastic/logstash#12503]
`MacOS 10.15.7`
`docker desktop community 2.5.0.1`

`docker pull docker.elastic.co/logstash/logstash:7.10.0`

`docker run --name=logstash-7.10 -v /xxx/logstash.yml:/usr/share/logstash/config/logstash.yml -v /xxx/logstash.conf:/usr/share/logstash/pipeline/logstash.conf -d docker.elastic.co/logstash/logstash:7.10.0`

- Mount external configuration file logstash.conf（utf-8）
 ```
input {
  beats {
    port => 5044
  }
}

output {
  stdout {
    codec => rubydebug
  }
}

 ```
`Failed to start`

LogStash::PipelineAction::Create/pipeline_id:main, :exception=>"LogStash::ConfigurationError", :message=>"Expected one of [ \t\r\n], "#", "input", "filter", "output" at line 1, column 1 (byte 1)", :backtrace=>["/usr/share/logstash/logstash-core/lib/logstash/compiler.rb:32:in compile_imperative'", "org/logstash/execution/AbstractPipelineExt.java:184:in initialize'", "org/logstash/execution/JavaBasePipelineExt.java:69:in initialize'", "/usr/share/logstash/logstash-core/lib/logstash/java_pipeline.rb:47:in initialize'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline_action/create.rb:52:in execute'", "/usr/share/logstash/logstash-core/lib/logstash/agent.rb:365:in block in converge_state'"]}

**Comment by ewilansky** [comment]
I found that getting rid of the blank lines in logstash.conf resolves this issue:
```
input {
 beats {
   port => 5044
 }
}
output {
 stdout {
   codec => rubydebug
 }
}
```

**Comment by ewilansky** [comment]
I think the issue is that logstash isn't seeing the logstash.conf file. I ran into this so I changed my bind mounts so that the logstash.conf file is projected into /etc/logstash/conf.d/logstash.conf. I still use the pipeline directory to hold the input file for the pipeline so my volume mounts look like this:
```
 -v ./docker/logstash/config/logstash.yml:/usr/share/logstash/config/logstash.yml:ro 
 -v ./docker/logstash/logstash.conf:/etc/logstash/conf.d/logstash.conf
 -v ./docker/logstash/pipeline/:/usr/share/logstash/pipeline/
```
Notice that the bind mount directory path for logstash.conf is ./docker/logstash while the pipeline bind mount directory path projected into the container is ./docker/pipeline.
